traits/src/location.rs

```diff
@@ -46,77 +47,87 @@ pub trait Reserve {
 
 impl Reserve for MultiAsset {
 	fn reserve(&self) -> Option<MultiLocation> {
-		if let MultiAsset::ConcreteFungible { id, .. } = self {
-			id.chain_part()
+		if let Concrete(location) = &self.id {
+			location.chain_part()
 		} else {
 			None
 		}
 	}
 }
 
+pub trait RelativeLocations {
+	fn sibling_parachain_general_key(para_id: u32, general_key: Vec<u8>) -> MultiLocation;
+}
+
+impl RelativeLocations for MultiLocation {
+	fn sibling_parachain_general_key(para_id: u32, general_key: Vec<u8>) -> MultiLocation {
+		MultiLocation::new(1, X2(Parachain(para_id), GeneralKey(general_key)))
+	}
+}
+
 #[cfg(test)]
 mod tests {
 	use super::*;
 
 	const PARACHAIN: Junction = Parachain(1);
-	const GENERAL_INDEX: Junction = GeneralIndex { id: 1 };
+	const GENERAL_INDEX: Junction = GeneralIndex(1);
 
 	fn concrete_fungible(id: MultiLocation) -> MultiAsset {
-		MultiAsset::ConcreteFungible { id, amount: 1 }
+		(id, 1).into()
 	}
 
 	#[test]
 	fn parent_as_reserve_chain() {
 		assert_eq!(
-			concrete_fungible(MultiLocation::X2(Parent, GENERAL_INDEX)).reserve(),
-			Some(Parent.into())
+			concrete_fungible(MultiLocation::new(1, X1(GENERAL_INDEX))).reserve(),
+			Some(MultiLocation::parent())
 		);
 	}
 
 	#[test]
 	fn sibling_parachain_as_reserve_chain() {
 		assert_eq!(
-			concrete_fungible(MultiLocation::X3(Parent, PARACHAIN, GENERAL_INDEX)).reserve(),
-			Some((Parent, PARACHAIN).into())
+			concrete_fungible(MultiLocation::new(1, X2(PARACHAIN, GENERAL_INDEX))).reserve(),
```

Comment:
You could use the RelativeLocations trait here

Reply:
I think this is makes a bit easier to reason the test